### DISCUSSION TITLE
is potential positive on rna?

### DISCUSSION THREAD

**Discussion by mishaguy** [Electrostatics/pdb2pqr#285] [General]
Hell All,

Recently pdb2pqr developers found the issue with rna from pdb - 4e8m. There were errors in processing this structure.
They seem to resolve this issue now.

I've decided to compare potential surface on this structure obtained by APBS in PyMol for two cases.
The first is when I do not prepare molecule at all. The surface looks mostly neutral (please see the fist attached screenshot).
The second screenshot is the surface built after pdb2pqr. It looks mostly positive. I used default Amber in PyMol but I think the similar results are now on webserver too.
Looks like it adds mostly positive ions, right?
Is this correct? Anyone comment on this would be valueless.
Thanks,
Misha Guy
Yale
<img width="567" alt="Screen Shot 2022-01-01 at 9 33 15 PM" src="https://user-images.githubusercontent.com/86742487/147865089-b968d6d0-78d4-4228-881d-47defde84365.png">
<img width="588" alt="Screen Shot 2022-01-01 at 9 38 47 PM" src="https://user-images.githubusercontent.com/86742487/147865091-d15190f1-7c04-4781-8c02-efd63c6f07fd.png">

**Comment by sobolevnrm** [comment]
Can you please provide step-by-step instructions on how to reproduce each of the images you have posted above?

**Comment by mishaguy** [comment]
Before providing my steps, I want to make sure that current version of APBS plugin is working in your PyMol.
If it is not, get back to me and I'll show how to get it working from current public distribution of PyMol.
ok, here we go.

1. Open PyMol and fetch 4e8m.
2. Click on A button on the right and select Remove waters.
3. Select APBS plugin from menu and uncheck Prepare Molecule. See first attached screenshot please.
<img width="552" alt="Screen Shot 2022-01-02 at 5 17 14 AM" src="https://user-images.githubusercontent.com/86742487/147872789-48ca1cfa-6e23-45f2-b179-a464074ccc6b.png">

4. Click Run, wait couple minutes and see the green ESP surface like on the second screenshot. I believe this is close to neutral.
<img width="546" alt="Screen Shot 2022-01-02 at 5 19 18 AM" src="https://user-images.githubusercontent.com/86742487/147872836-14272e42-53b3-4d9a-8868-ad0d21358cfd.png">

5. Type delete all in PyMol prompt, fetch 4e8m again, remove waters again (or just delete esp object).
6. Now keep Prepare Molecule checked and use your pdb2pqr (this is the latest 3.5.0, thanks to you), See attached screen
<img width="545" alt="Screen Shot 2022-01-02 at 5 22 21 AM" src="https://user-images.githubusercontent.com/86742487/147872890-0c12560d-199b-429e-ba36-ecf08eab9f8c.png">

7. Run and get this mostly red surface
<img width="598" alt="Screen Shot 2022-01-02 at 5 27 22 AM" src="https://user-images.githubusercontent.com/86742487/147872981-770dcecb-7c11-4b63-8ef9-01f1b704a4bf.png">

8. Now do the same choosing formal charge and vdw
<img width="563" alt="Screen Shot 2022-01-02 at 5 28 31 AM" src="https://user-images.githubusercontent.com/86742487/147873010-26acdbb3-89d5-4875-ba3d-e9454fc0e5de.png">

9. This is the result
<img width="597" alt="Screen Shot 2022-01-02 at 5 32 39 AM" src="https://user-images.githubusercontent.com/86742487/147873085-11fa5b8e-8687-4f2f-9cd9-cd67e94e102a.png">

Visually there is no much difference in surface built with preparing molecule by two methods. I am sure that if we go inside, there are numerical differences. However, these positive surfaces are far from almost neutral one built on structure like it came from pdb.
(I can check it with Gaussian, if we need to make sure it is quite correct).

This is my question, ok, I can understand that we restore some positive ions one way or another. By placing these heavy atoms I dramarically change the whole ESP field. How do we know that this is correct?

Couple months ago I had a researcher asking her to help to get EPS values close to specific atoms from such field. Thes atoms are in original structure, they are not restored. So the question raises should she use neutral surface or one with restored positivity.

The bottom line, I do not understand these results now and would appreciate anyone explaining this to me.

Best,
Misha

**Comment by sobolevnrm** [comment]
When I run Steps 1-4 above in PyMOL 2.4.1, I get a picture that is not a green surface:

![Screenshot 2022-01-02 184111](https://user-images.githubusercontent.com/1933272/147896413-640c8b4a-1662-4c53-8d61-222d840b73ae.png)

I don't know what to make of this step other than it appears PyMOL generates its own PQR file if PDB2PQR is not used to prepare the molecule.  @speleo3 do you have any insight?

I'll work on the next step with the new PDB2PQR 3.5.0 next.

**Comment by sobolevnrm** [comment]
To make Steps 5-7 work, I had to change the PDB2PQR command line options to `--ff=AMBER --keep-chain` which gave me the following picture that looks reasonable for RNA:

![image](https://user-images.githubusercontent.com/1933272/147896809-9a1b1c8e-05c7-4de7-8cee-f125be635bd4.png)

As expected the surface is very negative.  Note that the cations from the original PDB structure are not present in the PQR, presumably because PDB2PQR did not match the `K` and `MG` residues in the input PDB with known atom types (the `EPE` and `SPM` molecules are also not present in the PQR).  This could be easily addressed by manually adding entries for these atoms back into the PQR file produced by PDB2PQR.  Therefore, the following picture is much more accurate:

![image](https://user-images.githubusercontent.com/1933272/147897104-f40cfe16-e4d2-40ef-861d-f9b46f0a8cb6.png)

**Reply by sobolevnrm** [reply]
Note that I revised the picture/comment in this response to remove the representations of atoms that are not in the PQR.

**Comment by mishaguy** [comment]
I run PyMol available from public domain. When late DeLano sold it to Schrodinger, they were obligated to release the code all the time. My current versions are 2.5.0.
I see green surface on Mac and Windows. I do not run on Linux.
Where do you run it?
Also, do you have Schrodinger's site on your system?
Their formal version comes with APBS plugin where they have their own preparation of molecule but it calls Maestro or whatever it is called now, I do not have it on my computers with PyMol. We do have it on our HPC cluster and potentially I can run it ro prepare molecule outside and then to input to APBS.
I have a feeling that in your case, some kind of preparation happening.

**Reply by sobolevnrm** [reply]
I run Schrodinger's version of PyMOL on Windows.

**Comment by mishaguy** [comment]
First of all - I made a mistake saying it is positive surface, I interpreted red color as positive by accident. Sorry for this.
Still the question remains why restoring missing atoms changes surface so dramatically and is this correct physically?
This is the concern.
How do you add these previous atoms back to pqr?
thanks,

**Reply by sobolevnrm** [reply]
Please be more precise:  what "missing atoms" are you referring to and what changes are you describing?

**Comment by sobolevnrm** [comment]
When I run Steps 8-9, I get the following picture which also looks reasonable for RNA:

![image](https://user-images.githubusercontent.com/1933272/147897271-e99dc0af-969c-4760-9b5f-508c575fd193.png)

It looks like the ions and small molecules are also not included in the PQR file with this preparation approach:

![image](https://user-images.githubusercontent.com/1933272/147897544-e17b4837-ec4f-42fe-b4dc-cb2bdbc601ca.png)

In summary, I'm not very surprised that both surfaces are overwhelmingly negative given the RNA phosphate backbone and it's major influence on nucleic acid electrostatics.

**Comment by mishaguy** [comment]
yes, I figured how to do this manual step. you just hide original object in the right panel of PyMol.
<img width="901" alt="Screen Shot 2022-01-02 at 10 33 47 PM" src="https://user-images.githubusercontent.com/86742487/147898018-b9e0f3be-6612-4e64-b6c2-20d474e3a8bd.png">

**Comment by mishaguy** [comment]
Basically, when you prepare molecule, you sort of restoring 'missing' parts of structure based on original field, right? This is what I meant, sorry for not being more specific.
What interesting here to me is that you pdb2pqr for 4e8m produces the same result as using formal charge and vdw. This is convincing.
Now, I can run this formal charge and vdw even on other rna which you can't handle for now. I think this is the way to go.

ok, I got your argument about backbone being negative. this might address my concern.
at the same time, there was an old publication by Anna Pyle's group about using nonlinear Poisson-Boltzmann instead of linear.

http://crystal.med.upenn.edu/sharp-lab-pdfs/rna_elect.pdf

Do you use linear or nonlinear equation?

**Reply by sobolevnrm** [reply]
> Basically, when you prepare molecule, you sort of restoring 'missing' parts of structure based on original field, right? This is what I meant, sorry for not being more specific. 

PDB2PQR adds hydrogens and fills in a small number of heavy atoms if they are missing from the structure.  However, the influence of the missing heavy atoms on the surface potential is likely minimal (there are relatively few in 4E8M) and the sum of the hydrogen charges with the heavy atom charges are likely similar to the more coarse-grained charges that PDB2PQR is assigning.

> What interesting here to me is that you pdb2pqr for 4e8m produces the same result as using formal charge and vdw. This is convincing. Now, I can run this formal charge and vdw even on other rna which you can't handle for now. I think this is the way to go.

The influence of the high negative charge density from the RNA phosphates likely dominates the results for both preparation methods.

> ok, I got your argument about backbone being negative. this might address my concern. at the same time, there was an old publication by Anna Pyle's group about using nonlinear Poisson-Boltzmann instead of linear.
> 
> http://crystal.med.upenn.edu/sharp-lab-pdfs/rna_elect.pdf
> 
> Do you use linear or nonlinear equation?

Please read more about APBS at https://apbs.readthedocs.io/en/latest/ -- particularly https://apbs.readthedocs.io/en/latest/using/input/old/elec/npbe.html?highlight=npbe

**Comment by mishaguy** [comment]
ok, very good.
now I ran with npbe and this is what I see. 
<img width="863" alt="Screen Shot 2022-01-02 at 11 14 15 PM" src="https://user-images.githubusercontent.com/86742487/147899304-4de2e824-bf16-469f-8234-914f7a303f33.png">

**Comment by mishaguy** [comment]
if you want to repeat it in PyMol, to set npbe you change template in
C:\Users\Misha\AppData\Local\Programs\Python\Python38\Lib\site-packages\pmg_tk\startup\apbs_gui\electrostatics.py

rotate this object

**Comment by mishaguy** [comment]
Hi Nathan,
Do you have APBS exe for Win with compiled with multithread?
Also do you happen to know where to get APBS 3.0 for Mac? I got from Homebrew but it is 1.5.
Thanks,
Misha

**Reply by sobolevnrm** [reply]
This is a new question and really should be in a new discussion.

See https://github.com/Electrostatics/apbs/releases per https://apbs.readthedocs.io/en/latest/getting/index.html#installing-from-pre-compiled-binaries.

**Comment by mishaguy** [comment]
thanks, Nathan, I'll start new discussion about executable. May be someone has already built files.

In the mean time, for curiosity, I ran npbe on original, not restored 4e8m. It runs much longer in this case, unbelievably long comparing with restored molecule (that's why I asked for multithreading exe).
In any case, results looks the same as with linear equation. All green. 
<img width="592" alt="Screen Shot 2022-01-03 at 8 52 47 AM" src="https://user-images.githubusercontent.com/86742487/147939270-82fb188e-e095-4ba4-98a1-86444e8129db.png">

Best,
Misha

**Reply by sobolevnrm** [reply]
If the surface is all green, then you are doing something wrong.  See the discussion above.
